exporter/collectors/test_collector.py

```diff
@@ -50,27 +51,29 @@ def parse_test_response(use_v1_uptime_endpoints, tests, m_test_id_flat_list):
 
 class TestCollector(object):
 
-    def __init__(self, use_v1_uptime_endpoints, use_v1_maintenance_windows_endpoints, username, api_key, tags):
+    def __init__(self, use_v1_uptime_endpoints, use_v1_maintenance_windows_endpoints, timeout, username, api_key, tags):
         self.use_v1_uptime_endpoints = bool(distutils.util.strtobool(use_v1_uptime_endpoints))
         self.use_v1_maintenance_windows_endpoints = bool(distutils.util.strtobool(use_v1_maintenance_windows_endpoints))
+        self.timeout = timeout
         self.username = username
         self.api_key = api_key
         self.tags = tags
 
     def collect(self):
-
+        start_time = time.time()
         logger.info("Collector started.")
 
         try:
-
             maintenance = m.get_maintenance(self.use_v1_maintenance_windows_endpoints, self.api_key, self.username)
             try:
-                maintenance_data = maintenance.json()['data']
+                maintenance_data = []
+                if maintenance != "" and maintenance is not None:
+                    maintenance_data = maintenance.json()['data']
+                logger.info(f"Maintenance response:\n{maintenance_data}")
             except Exception as e:
                 logger.error(f"Could not parse maintenace data, exception: {e}")
                 logger.error(f"Data was:\n{maintenance}")
-                sys.exit(1)
-            logger.debug(f"Maintenance response:\n{maintenance_data}")
+                ##sys.exit(1)
```

Comment:
Is this intentional?

Reply:
We made it intentional as we do not want the script to stop when it errors while fetching maintenance data. We want it to proceed and read/yield the metric for all the tests from the status cake API.